### PR TITLE
Allow erc20 token address in full or lower letter case

### PR DESCRIPTION
**Pull request by jubalm**
Screenshot 2024-08-08 at 3 21 52 PM

### CODE REVIEW

**Review by KillariDev — APPROVED** [review]
image

This allows you to add a token with a wrong checksum thought, as it seems like checksum check is not done

**Comment by jubalm** [comment]
> image
> This allows you to add a token with a wrong checksum thought, as it seems like checksum check is not done

This should be fixed with [7feda45](https://github.com/DarkFlorist/lunaria/pull/278/commits/7feda457a65305a731ebb97e5ab9b0d958bca95e)

File: app/ts/components/TokenAdd.tsx [hunk 1]
```diff
@@ -205,16 +207,24 @@ const QueryStatus = () => {
 				</div>
 			)
 		case 'resolved':
-			const token = serialize(ERC20Token, tokenQuery.value.value)
+			// queried token will have included a checksum address
+			const fetchedToken = tokenQuery.value.value
+			const showChecksumWarning = useComputed(() => {
+				const userInputAddress = noHexPrefix(address.value)
```
Comment (KillariDev) [inline app/ts/components/TokenAdd.tsx:213]: why do you remove the hex prefix? this also fails to remove `0X`

**Review by KillariDev — APPROVED** [review]

File: app/ts/components/TokenAdd.tsx (hunk 1 above)
Comment (jubalm) [reply app/ts/components/TokenAdd.tsx]: it's now using ethersjs checksum validation which should already do this actually

**Comment by jubalm** [comment]
The fields should now correctly check for address checksum and accepts all lower-case or upper-case addresses

| Screenshot 2024-09-20 at 8 44 34 PM | image |
|--|--|
| *token add* | *send* |

**Review by KillariDev — APPROVED** [review]

**Pull request merged**